model/annotations/task_annotations.go

```diff
@@ -204,6 +204,49 @@ func UpdateAnnotation(a *TaskAnnotation, userDisplayName string) error {
 	return errors.Wrapf(err, "problem adding task annotation for '%s'", a.TaskId)
 }
 
+// InsertManyAnnotations updates the source for a list of task annotations and their issues and then inserts them into the DB.
+func InsertManyAnnotations(annotations []TaskAnnotation, userDisplayName string) error {
+	source := &Source{
+		Author:    userDisplayName,
+		Time:      time.Now(),
+		Requester: APIRequester,
+	}
+	for i, _ := range annotations {
+		update := bson.M{}
+		if annotations[i].Metadata != nil {
+			update[MetadataKey] = annotations[i].Metadata
+		}
+		if annotations[i].Note != nil {
+			annotations[i].Note.Source = source
+			update[NoteKey] = annotations[i].Note
+		}
+		if annotations[i].Issues != nil {
+			for i := range annotations[i].Issues {
+				annotations[i].Issues[i].Source = source
+			}
+			update[IssuesKey] = annotations[i].Issues
+		}
+		if annotations[i].SuspectedIssues != nil {
+			for i := range annotations[i].SuspectedIssues {
+				annotations[i].SuspectedIssues[i].Source = source
+			}
+			update[SuspectedIssuesKey] = annotations[i].SuspectedIssues
+		}
+
+		_, err := db.Upsert(
```

Comment:
Do we maybe want to do this in one db call instead?

Reply:
Since the structure of the request [here](https://jira.mongodb.org/browse/EVG-16100?focusedCommentId=4286412&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-4286412) is that each task/execution pair is associated with a single annotation, I think the best I can do is bulk upserting the task/execution pairs for each annotation. I've done so